lightning-background-processor/src/lib.rs

```diff
@@ -928,14 +962,46 @@ where
 			Some(false) => {
 				log_trace!(logger, "Regenerating sweeper spends if necessary");
 				if let Some(ref sweeper) = sweeper {
-					let _ = sweeper.regenerate_and_broadcast_spend_if_necessary().await;
+					let sweeper = sweeper.clone();
+					let fut: Pin<
+						Box<
+							dyn Future<Output = Result<(), (lightning::io::Error, bool)>>
+								+ Send
+								+ 'static,
+						>,
+					> = Box::pin(async move {
+						sweeper.regenerate_and_broadcast_spend_if_necessary().await.map_err(|_| {
```

Comment:
Because this is a native async method, I don't think there was a way to avoid clone.